backend/src/packages/users/user.service.ts

```diff
@@ -95,8 +95,13 @@ class UserService implements Service {
     return Promise.resolve(null);
   }
 
-  public delete(): ReturnType<Service['delete']> {
-    return Promise.resolve(true);
+  public async delete(id: number): Promise<boolean> {
+    const user = await this.findById(id);
+    if (!user) {
+      return false;
+    }
+
+    return await this.userRepository.delete(id);
```

Comment:
```suggestion
    return user && await this.userRepository.delete(id);
```
Definitely if the linter allows it

Reply:
I could implement it like this to avoid type problems:

return user ? await this.userRepository.delete(id) : false;

 but I believe that for the reliability of the code is better to keep the return as simple as possible. 

Implementing it like this:

return user && await this.userRepository.delete(id);

 will mean changes in types, which if required is okay, but if it can be avoided is better to keep it simple since is just small code.

Reply:
If there's a problem with return type, you can add `Boolean(...)` to check if data exists